lms/services/lti_grading/_v13.py

```diff
@@ -62,6 +62,54 @@ def record_result(self, grading_id, score=None, pre_record_hook=None):
             headers={"Content-Type": "application/vnd.ims.lis.v1.score+json"},
         )
 
+    def read_lineitems(self, lineitems_url, resource_link_id=None):
+        """
+        Read all the lineitem present in the `lineitems_url` container.
+
+        https://www.imsglobal.org/spec/lti-ags/v2p0#container-request-filters
+
+        :param lineitems_url: URL of the lineitems container
+        :param resource_link_id: Filter lineitems only relevant for this assignment id.
+        """
+        params = {}
+
+        if resource_link_id:
+            params["resource_link_id"] = resource_link_id
+
+        return self._ltia_service.request(
+            "GET",
+            lineitems_url,
+            scopes=self.LTIA_SCOPES,
+            params=params,
+            headers={"Accept": "application/vnd.ims.lis.v2.lineitemcontainer+json"},
+        ).json()
```

Comment:
I think the only reason we read the line items is to indirectly check if the line item exists for a specific assignment before we create it.

So I think a few things are true:

 * We never want to call this without a `resource_link_id` - So it probably shouldn't be optional
 * We don't actually care about the response, although we aren't doing much conversion here
 * i.e. This could be `has_line_items` if all we care about is existence
 * I think this could also be a private method called from `create_lineitem` so that `create_lineitem` becomes `upsert_lineitem`. 

Also, what happens if you create a line item for an assignment without checking if one is there already. Can you end up with two, overwrite it, or does it ignore you?

Reply:
POST'ing multiple does create one per call. 

Removed the read_line_item method in any case in favor of just checking the presence of `lineitem` (via is_gradable_assignment)